### ISSUE TITLE
Question about lb_margin_th

### ISSUE THREAD

**Issue by Huage001** [yongchao97/FRePo#7]
Dear authors,

Thanks for your excellent work!

I have a question related to function *lb_margin_th*. As shown in the following line, if I understand correctly, val[0] and val[1] should have contained the top 2 values of logits. If this is the case, I think this line should be: 

margin = jnp.minimum(val[..., 0] - val[..., 1], 1 / dim) 

because the top_k function applies on the last axis by default.

Or is there anything I got wrong? Looking forward to your reply and thanks in advance!

https://github.com/yongchao97/FRePo/blob/43e028a5839a5de367701b9e9544a08ffccb3166/lib/datadistillation/frepo.py#L156

**Comment by Huage001** [comment]
Nothing, I noted that you use vmap.
